packages/plugins/plugin-table/src/components/Table/NewColumnForm.tsx

```diff
@@ -0,0 +1,76 @@
+//
+// Copyright 2024 DXOS.org
+//
+
+import * as ModalPrimitive from '@radix-ui/react-popper';
+import React, { useCallback, useEffect, useState } from 'react';
+
+import { create, createObjectId } from '@dxos/echo-schema';
+import { DropdownMenu, useThemeContext } from '@dxos/react-ui';
+import { Field } from '@dxos/react-ui-data';
+import { type FieldType, FieldValueType, getUniqueProperty, type ViewType } from '@dxos/schema';
+
+import { type TableModel } from '../../model';
+
+type NewColumnFormProps = {
+  tableModel?: TableModel;
+  open: boolean;
+  close: () => void;
```

Comment:
Generally try to make props optional, also onX standards.
Also define type immed. before usage in function.

```tsx
open?: boolean;
onClose?: () => void;
```